### PR TITLE
Fix visit_id range for JS limits.

### LINKED ISSUE #277: 64-bit integer columns truncated in Javascript
The max safe integer value in JS is [53 bits](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER). We generate 64-bit values for `visit_id` when the S3 aggregator is used, meaning some values get clipped when sent to the Javascript instrumentation. Since this same clipping doesn't happen in python, this means that the `visit_id` column is inconsistent between tables written directly in python (e.g., `site_visits`) and those written by the javascript instrumentation (e.g., `http_requests`). 

The simple solution here is to mask the [generated visit IDs](https://github.com/mozilla/OpenWPM/blob/b3ead7e38892095950806e8bcbb2e1129c27ca96/automation/DataAggregator/S3Aggregator.py#L320-L323) to 53 bits. We should also add a test for visit_id consistency across all tables when fixing this.

Note that the `expires` column of the `javascript_cookies` table is [also set to use 64 bits](https://github.com/mozilla/OpenWPM/blob/b3ead7e38892095950806e8bcbb2e1129c27ca96/automation/DataAggregator/parquet_schema.py#L149).

### PR DESCRIPTION
**Pull request by englehardt**
Will fix #277. Alternative to https://github.com/mozilla/OpenWPM/pull/323, which has errors related to signed/unsigned integers and does not pass the new included test. I verified that the master branch fails this test due to the truncation issue.

This also fixes two other issues:
* The S3 Aggregator would silently drop the last page visit (or set of visits).
* `utilities::get_test_config` accepts a browser count as an argument, but only makes the first browser headless.

### CODE REVIEW

**Comment by englehardt** [comment]
Should be rebased on #382

File: automation/DataAggregator/BaseAggregator.py [hunk 1]
```diff
@@ -93,10 +93,9 @@ def shutdown(self):
 
     def drain_queue(self):
         """ Ensures queue is empty before closing """
-        self.sock.close()
```
Comment (eriktews) [inline automation/DataAggregator/BaseAggregator.py]: Is that change intentional?

Comment (englehardt) [reply automation/DataAggregator/BaseAggregator.py]: Yes, we end up calling `sock.close()` twice. Once here and once in `BaseAggregator::close`.

File: automation/DataAggregator/S3Aggregator.py [hunk 2]
```diff
@@ -257,6 +257,8 @@ def process_content(self, record):
     def drain_queue(self):
         """Process remaining records in queue and sync final files to S3"""
         super(S3Listener, self).drain_queue()
+        for crawl_id in self.browser_map.keys():
```
Comment (nhnt11) [inline automation/DataAggregator/S3Aggregator.py]: Can we iterate on the values instead of the keys, and directly call `_create_batch` on these? Maybe there's a python idiom to do it all in one line?

File: test/openwpmtest.py [hunk 3]
```diff
@@ -40,7 +40,8 @@ def get_test_config(self, data_dir="",
             load_default_params(num_browsers)
         manager_params['data_directory'] = data_dir
         manager_params['log_directory'] = data_dir
-        browser_params[0]['headless'] = True
+        for i in range(num_browsers):
```
Comment (nhnt11) [inline test/openwpmtest.py]: If you think it makes sense, maybe add an OS check here and exclude this on darwin

File: automation/DataAggregator/S3Aggregator.py (hunk 2 above)
Comment (englehardt) [reply automation/DataAggregator/S3Aggregator.py]: I updated to iterating on values. I think we could also do `map(self._create_batch, self.browser_map.values())`, but I think using a loop that names the values (`visit_id`) makes it more understandable as to what we're looping over.

File: test/openwpmtest.py (hunk 3 above)
Comment (englehardt) [reply test/openwpmtest.py]: This is a good idea, but I think we should address it globally. I'll make a new PR that disables it in the browser deployment script.

**Comment by englehardt** [comment]
Rebased on #382, actually added the test page to the PR :), and addressed Nihanth's comments. Will merge if tests pass.

File: test/openwpmtest.py (hunk 3 above)
Comment (englehardt) [reply test/openwpmtest.py]: See #388
